packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -643,6 +642,30 @@ class HeapTreeViewState extends State<HeapTree>
     controller.monitorAllocations = currentAllocations;
   }
 
+  void highlightDropdown(bool directionDown) {
+    final maxItems = controller.searchAutoComplete.value.length;
+    var indexToSelect = controller.currentDefaultIndex;
+    if (directionDown) {
+      // Select next item in auto-complete overlay.
+      if (++indexToSelect >= maxItems) {
```

Comment:
nit: having side effects in an iff statement is a bit surprising. Maybe change to 
```dart
++indexToSelect;
if (indexToSelect >= maxItems)
```
so the code is more obvious to people reading it who might not notice the side effects.

Reply:
Yep, fixed.